activerecord/lib/active_record/connection_adapters/postgresql/schema_statements.rb

```diff
@@ -144,6 +144,24 @@ def indexes(table_name) # :nodoc:
           end
         end
 
+        # Returns an array of enum names defined in the database.
+        def enums
+          query(<<-SQL, "SCHEMA").flatten
+            SELECT DISTINCT t.typname AS enum_name
+            FROM pg_type t
+            JOIN pg_enum e ON t.oid = e.enumtypid
+            JOIN pg_catalog.pg_namespace n ON n.oid = t.typnamespace;
```

Comment:
Could this be returning types that are not enums?

Reply:
The join with `pg_enum` prevent that from happening.